kinder/hack/update-deps.sh

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env bash
+# Copyright 2018 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# Runs go mod tidy, go mod vendor, and then prun vendor
+#
+# Usage:
+#   update-deps.sh
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# shellcheck source=/dev/null
+source "$(dirname "$0")/utils.sh"
+# cd to the root path
+cd_root_path
+
+prune-vendor() {
+  find vendor -type f \
+    -not -iname "*.c" \
+    -not -iname "*.go" \
+    -not -iname "*.h" \
+    -not -iname "*.proto" \
+    -not -iname "*.s" \
+    -not -iname "AUTHORS*" \
+    -not -iname "CONTRIBUTORS*" \
+    -not -iname "COPYING*" \
+    -not -iname "LICENSE*" \
+    -not -iname "NOTICE*" \
+    -exec rm '{}' \;
+}
+
+export GO111MODULE="on"
+go mod tidy
+# TODO enable in kinder?
```

Comment:
Unless necessary, I think we should wait until kinder gets more stable (end of this cycle) before starting to vendor, so we can change without getting PR big.

Reply:
ack.